### DISCUSSION TITLE
How can I export documents to Word or PDF?

### DISCUSSION THREAD

**Discussion by junoriosity** [ueberdosis/tiptap#4242] [General Questions]
It would be great to export a document to Word resp. PDF.

However, I have not found a way to do this.

It would be great if you could help me here.

**Comment by collasanta** [comment]
You can use something like puppeteer.js on the backend, and applying the html from the document to be TipTap downloaded:

```
const browser = await puppeteer.launch();
const page = await browser.newPage();
await page.setContent(fullHtml, { waitUntil: 'networkidle0' });
const pdfBuffer = await page.pdf({ format: 'A4' });
await browser.close();
const response = new NextResponse(pdfBuffer);
```
They seem to support  word exports .docx for now
https://tiptap.dev/docs/editor/extensions/functionality/export

**Comment by waqarilyas** [comment]
Any update on this?

**Comment by david1542** [comment]
Would also love to export Tiptap documents to PDFs

**Comment by SawkaDev** [comment]
There are a few options:
1. Puppeteer as the first comment mentions
2. Browser Print API 
3. Pro Extension: Export to DOCX. However I have found tons of limitations and formatting issues with this, my hope is that they develop this extensions much more https://tiptap.dev/docs/editor/extensions/functionality/export
4. take the JSON output and traverse it yourself and use libraries as you wish (eg pdfmake for pdfs and docx for docx). This is what I ended up doing for documents that need a lot of design control and accuracy to match what the editor showed. But it is a pain to do and takes a lot of time.


I’m hoping Tiptap invests more into export. We should be able to take the editor WYSIWYG and export it 1:1 to a pdf or doc and still see the same WYSIWYG
